src/interpreter/contexts.jl

```diff
@@ -109,10 +99,126 @@ end
 
 function _is_primitive_expression(Tctx, Tmode, sig)
     return quote
-        function Mooncake.is_primitive(
+        function Mooncake._is_primitive(
             ::Type{$(esc(Tctx))}, ::Type{<:$(Tmode)}, ::Type{<:$(esc(sig))}
         )
             return true
         end
     end
 end
+
+const _IS_PRIMITIVE_CACHE = IdDict{Any,Bool}()
+
+"""
+    is_primitive(ctx::Type, mode::Type{<:Mode}, sig::Type{<:Tuple}, world::UInt)
+
+Returns a `Bool` specifying whether the methods specified by `sig` are considered primitives
+in the context of context `ctx` in mode `mode` at world age `world`.
+
+```jldoctest
+julia> using Mooncake: is_primitive, DefaultCtx, ReverseMode
+
+julia> is_primitive(DefaultCtx, ReverseMode, Tuple{typeof(sin), Float64}, Base.get_world_counter())
+true
+```
+
+`world` is needed as rules which Mooncake derives are associated to a particular Julia world
+age. As a result, anything declared a primitive after the construction of a rule ought not
+to be considered a primitive by that rule. To see how this works, consider the following:
```

Comment:
I'd clarify further by explaining how to override an existing rule from a world age where primitives are different. Feel free to edit the wording as you see fit. 

```suggestion
to be considered a primitive by that rule. One can explicitly derive a new rule via `build_rrule` after new functions are declared as primitive, should it be needed in cases where a rule has been previously derived. To see how this works, consider the following:
```